### PR TITLE
AUT-2932: Enable email address checks

### PR DESCRIPTION
**Pull request by mattvot**
## What

Enables the `support_email_check_enabled` feature switch in all environments which enables the API side logic for handling email checks during account creation and email address updates.

## How to review

1. Code Review

## Checklist

- [X] Impact on orch and auth mutual dependencies has been checked.

## Related PRs

https://github.com/govuk-one-login/authentication-frontend/pull/1898

### CODE REVIEW

**Comment by pskushwaha1** [comment]
@mattvot , Below variable in Send OTP Notification  repeated twice  
`SUPPORT_EMAIL_CHECK_ENABLED            = var.support_email_check_enabled`
**Here**
https://github.com/govuk-one-login/authentication-api/blob/2b60e89b8b7b830c750bedce4bb04d45098e7b44/ci/terraform/account-management/send_otp_notification.tf#L31C1-L31C5
**and Here** 
https://github.com/govuk-one-login/authentication-api/blob/2b60e89b8b7b830c750bedce4bb04d45098e7b44/ci/terraform/oidc/send_notification.tf#L36-L44

Also There a count flag for Variable  `var.support_email_check_enabled` this needs to be removed i Believe? 

https://github.com/govuk-one-login/authentication-api/blob/2b60e89b8b7b830c750bedce4bb04d45098e7b44/ci/terraform/utils/email_check_results_writer_lambda.tf#L2

**Review by pskushwaha1 — DISMISSED** [review]

**Comment by mattvot** [comment]
Looks like `deploy_check_email_fraud_block_count` also needs updating
https://github.com/govuk-one-login/authentication-api/blob/698673e5754e829ecaf35688ef617ec79c6834d6/ci/terraform/oidc/site.tf#L47

File: ci/terraform/oidc/site.tf [hunk 1]
```diff
@@ -44,7 +44,7 @@ locals {
   deploy_account_interventions_count   = 1
   deploy_ticf_cri_count                = contains(["sandpit", "authdev1", "authdev2", "dev", "build", "staging", "integration"], var.environment) ? 1 : 0
   deploy_reauth_user_count             = contains(["build", "sandpit", "authdev1", "authdev2", "staging", "integration"], var.environment) ? 1 : 0
-  deploy_check_email_fraud_block_count = contains(["build", "sandpit", "authdev1", "authdev2", "staging"], var.environment) ? 1 : 0
+  deploy_check_email_fraud_block_count = contains(["build", "sandpit", "authdev1", "authdev2", "staging", "integration", "production"], var.environment) ? 1 : 0
```
Comment (mattvot) [inline ci/terraform/oidc/site.tf:47]: @pskushwaha1, these are the correct new values right?

Ideally I'd remove all the flags but this got very fiddly with the integration tests, so flag removal will have to follow later.

Pretty much done in https://github.com/govuk-one-login/authentication-api/pull/4999

Comment (pskushwaha1) [reply ci/terraform/oidc/site.tf:47]: As discussed we can go with` deploy_check_email_fraud_block_count = 1`

**Review by pskushwaha1 — APPROVED** [review]

**Pull request merged**